k8s/jenkins/legion-jenkins-plugin/src/main/resources/legion.groovy

```diff
@@ -80,11 +119,21 @@ def pod(Map podParams=null, Closure body) {
             "AIRFLOW_S3_URL", "AIRFLOW_REST_API", "AIRFLOW_DAGS_DIRECTORY", "DAGS_VOLUME_PVC"
     ]
 
-    envVars = envToPass.collect({ name -> envVar(key: name, value: System.getenv(name)) })
-    envVars << envVar(key: 'ENCLAVE_DEPLOYMENT_PREFIX', value: "${env.ENCLAVE_DEPLOYMENT_PREFIX}")
-    envVars << envVar(key: 'MODEL_SERVER_URL', value: "http://${env.ENCLAVE_DEPLOYMENT_PREFIX}${params.Enclave}-edge.${params.Enclave}")
-    envVars << envVar(key: 'EDI_URL', value: "http://${env.ENCLAVE_DEPLOYMENT_PREFIX}${params.Enclave}-edi.${params.Enclave}")
+    // Collect list of ENV variables to pass to slave instance
+    envVarsList = envToPass.collect({ name -> envVar(key: name, value: System.getenv(name)) })
+    envVarsList << envVar(key: 'ENCLAVE', value: "${enclave}")
+    envVarsList << envVar(key: 'ENCLAVE_DEPLOYMENT_PREFIX', value: "${env.ENCLAVE_DEPLOYMENT_PREFIX}")
+    envVarsList << envVar(key: 'MODEL_SERVER_URL', value: "http://${env.ENCLAVE_DEPLOYMENT_PREFIX}${enclave}-edge.${enclave}")
```

Comment:
@kirillmakhonin urls templates should be defined in configuration but not hardcoded in groovy pipeline lib script